src/Command/CommandHelper.php

```diff
@@ -326,4 +337,59 @@ public static function listRepositories(Composer $composer, OutputInterface $out
             ));
         }
     }
+
+    public static function handlePackageNotFound(
+        InvalidPackageName|UnableToResolveRequirement $exception,
+        FindMatchingPackages $findMatchingPackages,
+        OutputInterface $output,
+        TargetPlatform $targetPlatform,
+        ContainerInterface $container,
+    ): int {
+        $pieComposer = PieComposerFactory::createPieComposer(
+            $container,
+            PieComposerRequest::noOperation(
+                $output,
+                $targetPlatform,
+            ),
+        );
+
+        $package = $exception->requestedPackageAndVersion->package;
+        if (str_starts_with($package, 'ext-')) {
+            $package = substr($package, 4);
+        }
+
+        $output->writeln('');
+        $output->writeln(sprintf('<error>Could not install package: %s</error>', $package));
+        $output->writeln($exception->getMessage());
+
+        try {
+            $matches = $findMatchingPackages->for($pieComposer, $package);
+
+            if (count($matches)) {
+                $output->writeln('');
+                if (count($matches) === 1) {
+                    $output->writeln('<info>Did you mean this?</info>');
+                } else {
+                    $output->writeln('<info>Did you mean one of these?</info>');
+                }
+
+                array_map(
+                    static function (array $match) use ($output): void {
+                        $output->writeln(sprintf(' - %s: %s', $match['name'], $match['description'] ?? 'no description available'));
```

Comment:
Do you have access to the `php-ext.extension-name` from the `composer.json` of the extension to show the provided extension name? It could be displayed here.